app/javascript/components/Header/index.js

```diff
@@ -1,40 +1,237 @@
-import React, { useContext } from 'react'
-
-import { Link } from 'react-router'
+import React, { useState, useContext } from 'react'
+import { Link, withRouter } from 'react-router'
 import * as R from 'ramda'
-import Popover from 'material-ui/Popover'
-import Menu from 'material-ui/Menu'
-import MenuItem from 'material-ui/MenuItem'
-import Divider from 'material-ui/Divider'
-import ArrowDropRight from 'material-ui/svg-icons/navigation-arrow-drop-right'
+import styled from 'styled-components'
 import { useMutation } from '@apollo/react-hooks'
+import { useTranslation } from 'react-i18next'
+
+import {
+  Dropdown,
+  Trigger,
+  Menu as GardenMenu,
+  Item,
+  Separator,
+  PreviousItem,
+  HeaderItem,
+  NextItem,
+} from '@zendeskgarden/react-dropdowns'
+import { Avatar } from '@zendeskgarden/react-avatars'
+import TranslationIcon from '@zendeskgarden/svg-icons/src/12/translation-exists-fill.svg'
+import { MD } from '@zendeskgarden/react-typography'
 
-import ContentIcon from 'material-ui/svg-icons/content/create'
 import UpdateUserOfficeMutation from '/mutations/UpdateUserOfficeMutation.gql'
 import { UserContext, FilterContext } from '/context'
-import { present } from '../../lib/utils'
-import s from './main.css'
 
-const styles = {
-  item: {
-    fontSize: 14,
-  },
+const NavigationText = styled.div`
+  color: #08c;
+`
+
+const UserDetails = styled.div`
+  margin-left: 10px;
+`
+
+const UserProfileContainer = styled.button`
+  display: flex;
+  border: none;
+  text-align: left;
+`
+
+const UserName = styled(MD)`
+  font-weight: bold;
+`
+
+const UserProfile = ({ currentUser, offices, handleOfficeSelect, location, router }) => {
+  const { i18n, t } = useTranslation()
+  const languages = [
+    { label: 'English', value: 'en' },
+    // Enable when Japanense is supported
+    // { label: '日本語', value: 'ja' }
+  ]
+
+  const [isOpen, setIsOpen] = useState(false)
+  const [tempSelectedItem, setTempSelectedItem] = useState()
+  const [selectedItem, setSelectedItem] = useState({ office: currentUser.office, language: languages[0] })
+
+  const handleStateChange = (changes, stateAndHelpers) => {
+    if (R.hasPath(['isOpen'])(changes)) {
+      setIsOpen(
+        changes.selectedItem === 'default-office' ||
+          changes.selectedItem === 'language-settings' ||
+          changes.selectedItem === 'general-settings' ||
+          changes.isOpen
```

Comment:
Can we not just check if `changes.selectedItem` has a value? when would it be set to anything but these three?

Reply:
changes.selectedItem also could be a menu item, ie. 'English' (an item of languages) in which we would want to close the menu, as opposed selecting a sub-menu.